org.eclipse.xtext.util/src/org/eclipse/xtext/util/formallang/PdaUtil.java

```diff
@@ -772,7 +772,7 @@ public <S, P, D extends Pda<S, P>> D filter(Pda<S, P> pda, Predicate<S> filter,
 			LinkedList<S> todo = Lists.newLinkedList();
 			todo.add(orig);
 			Set<S> visited = Sets.newHashSet();
-			Set<S> folowers = Sets.newHashSet();
+			Set<S> folowers = Sets.newLinkedHashSet();
```

Comment:
Maybe fix the typo in the variable name since you're at it? Not strictly necessary though.

Certainly a candidate for the unstable order since we expose the followers. Nice catch!